cloud/amazon/iam_policy.py

```diff
@@ -183,6 +183,14 @@ def role_action(module, iam, name, policy_name, skip, pdoc, state):
     current_policies = [cp for cp in iam.list_role_policies(name).
                                         list_role_policies_result.
                                         policy_names]
+  except boto.exception.BotoServerError as e:
+    if e.error_code == "NoSuchEntity":
+      # Role doesn't exist so it's safe to assume the policy doesn't either
```

Comment:
@defionscode - Yep, what @tcable said.  An inline policy can not exist without the role.